docs/proposal/observability.md

```diff
@@ -0,0 +1,305 @@
+// default metrics.
+MetricSelector_ALL_METRICS MetricSelector_IstioMetric = 0
+// Counter of requests to/from an application, generated for HTTP, HTTP/2,
+// and GRPC traffic.
+//
+// The Prometheus provider exports this metric as: `istio_requests_total`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/request_count` (SERVER mode)
+// - `istio.io/service/client/request_count` (CLIENT mode)
+MetricSelector_REQUEST_COUNT MetricSelector_IstioMetric = 1
+// Histogram of request durations, generated for HTTP, HTTP/2, and GRPC
+// traffic.
+//
+// The Prometheus provider exports this metric as:
+// `istio_request_duration_milliseconds`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/response_latencies` (SERVER mode)
+// - `istio.io/service/client/roundtrip_latencies` (CLIENT mode)
+MetricSelector_REQUEST_DURATION MetricSelector_IstioMetric = 2
+// Histogram of request body sizes, generated for HTTP, HTTP/2, and GRPC
+// traffic.
+//
+// The Prometheus provider exports this metric as: `istio_request_bytes`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/request_bytes` (SERVER mode)
+// - `istio.io/service/client/request_bytes` (CLIENT mode)
+MetricSelector_REQUEST_SIZE MetricSelector_IstioMetric = 3
+// Histogram of response body sizes, generated for HTTP, HTTP/2, and GRPC
+// traffic.
+//
+// The Prometheus provider exports this metric as: `istio_response_bytes`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/response_bytes` (SERVER mode)
+// - `istio.io/service/client/response_bytes` (CLIENT mode)
+MetricSelector_RESPONSE_SIZE MetricSelector_IstioMetric = 4
+// Counter of TCP connections opened over lifetime of workload.
+//
+// The Prometheus provider exports this metric as:
+// `istio_tcp_connections_opened_total`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/connection_open_count` (SERVER mode)
+// - `istio.io/service/client/connection_open_count` (CLIENT mode)
+MetricSelector_TCP_OPENED_CONNECTIONS MetricSelector_IstioMetric = 5
+// Counter of TCP connections closed over lifetime of workload.
+//
+// The Prometheus provider exports this metric as:
+// `istio_tcp_connections_closed_total`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/connection_close_count` (SERVER mode)
+// - `istio.io/service/client/connection_close_count` (CLIENT mode)
+MetricSelector_TCP_CLOSED_CONNECTIONS MetricSelector_IstioMetric = 6
+// Counter of bytes sent during a response over a TCP connection.
+//
+// The Prometheus provider exports this metric as:
+// `istio_tcp_sent_bytes_total`.
+//
+// The Stackdriver provider exports this metric as:
+//
+// - `istio.io/service/server/sent_bytes_count` (SERVER mode)
+// - `istio.io/service/client/sent_bytes_count` (CLIENT mode)
```

Comment:
Stackdriver support will be removed from OSS istio in 1.23, you should remove all about it.